runtime/tests/third_party/dummy_device.toml

```diff
@@ -1,110 +1,104 @@
+Toffoli = { controllable = true }
+CY = { }
+CZ = { }
+PhaseShift = { controllable = true }
+ControlledPhaseShift = { }
+RX = { controllable = true }
+RY = { controllable = true }
+RZ = { controllable = true }
+Rot = { controllable = true }
+CRX = { }
+CRY = { }
+CRZ = { }
+CRot = { }
+Identity = { controllable = true }
+IsingXX = { controllable = true }
+IsingYY = { controllable = true }
+IsingZZ = { controllable = true }
+IsingXY = { controllable = true }
 
-# The union of all gate types listed in this section must match what
-# the device considers "supported" through PennyLane's device API.
-[[operators.gates]]
-native = [
-        "QubitUnitary",
-        "PauliX",
-        "PauliY",
-        "PauliZ",
-        "MultiRZ",
-        "Hadamard",
-        "S",
-        "T",
-        "CNOT",
-        "SWAP",
-        "CSWAP",
-        "Toffoli",
-        "CY",
-        "CZ",
-        "PhaseShift",
-        "ControlledPhaseShift",
-        "RX",
-        "RY",
-        "RZ",
-        "Rot",
-        "CRX",
-        "CRY",
-        "CRZ",
-        "CRot",
-        "Identity",
-        "IsingXX",
-        "IsingYY",
-        "IsingZZ",
-        "IsingXY",
-]
+[operators.gates.decomp]
 
 # Operators that should be decomposed according to the algorithm used
 # by PennyLane's device API.
 # Optional, since gates not listed in this list will typically be decomposed by
 # default, but can be useful to express a deviation from this device's regular
 # strategy in PennyLane.
-decomp = [
-        "SX",
-        "ISWAP",
-        "PSWAP",
-        "SISWAP",
-        "SQISW",
-        "CPhase",
-        "BasisState",
-        "QubitStateVector",
-        "StatePrep",
-        "ControlledQubitUnitary",
-        "DiagonalQubitUnitary",
-        "SingleExcitation",
-        "SingleExcitationPlus",
-        "SingleExcitationMinus",
-        "DoubleExcitation",
-        "DoubleExcitationPlus",
-        "DoubleExcitationMinus",
-        "QubitCarry",
-        "QubitSum",
-        "OrbitalRotation",
-        "QFT",
-        "ECR",
-]
+SX = {}
+ISWAP = {}
+PSWAP = {}
+SISWAP = {}
+SQISW = {}
+CPhase = {}
+BasisState = {}
+QubitStateVector = {}
+StatePrep = {}
+ControlledQubitUnitary = {}
+DiagonalQubitUnitary = {}
+SingleExcitation = {}
+SingleExcitationPlus = {}
+SingleExcitationMinus = {}
+DoubleExcitation = {}
+DoubleExcitationPlus = {}
+DoubleExcitationMinus = {}
+QubitCarry = {}
+QubitSum = {}
+OrbitalRotation = {}
+QFT = {}
+ECR = {}
 
 # Gates which should be translated to QubitUnitary
-matrix = [
-        "MultiControlledX",
-]
+[operators.gates.matrix]
+
+MultiControlledX = {}
+
+
+[operators.observables]
+# Observables supported by the device
+
+PauliX = {}
+PauliY = {}
+PauliZ = {}
+Hadamard = {}
+Hermitian = {}
+Identity = {}
+Projector = {}
+SparseHamiltonian = {}
+Hamiltonian = {}
+Sum = {}
+SProd = {}
+Prod = {}
+Exp = {}
+
+[measurement_processes.exactshots]
+
+Expval = {}
+Var = {}
+Probs = {}
+Stat = {}
+
+[measurement_processes.finiteshots]
```

Comment:
We could exact and finite shots properties of the entries instead of tables, what do you think? Avoids duplicating them

Reply:
@dime10 do I understand it correctly, that exactshots (mysterious, indeed) means that these observables are supported for `shots=None` programs? In general, yes, I agree that it would be better to encode this information as attributes.

Reply:
Yes exactly 👍

Reply:
Ok, will switch to the attribute style with `analytic` flag.